### DISCUSSION TITLE
Fully Disable Beeps?

### DISCUSSION THREAD

**Discussion by embraceentropy** [Jyers/Marlin#745] [Q&A] [answered]
I use/love the LCD beeper disable option...and my dog thanks you! However, I still get beeps when saving, erasing, storing settings and I get ~5 beeps during the M600 filament change or initiating filament change using the LCD when the LCD is waiting on input. My dog does not like beeps and reacts to them like most dogs react to thunder (at least it keeps us honest with smoke detector batteries). 

I have changed the following parameters trying to disable all beeps (even ones I don't know about yet). Besides de-soldering the speaker are there other parameters I can change to fully disable the speaker?

#if ENABLED(BEEP_ON_FEEDRATE_CHANGE)
  #define FEEDRATE_CHANGE_BEEP_DURATION   0
  #define FEEDRATE_CHANGE_BEEP_FREQUENCY 440

#define LCD_FEEDBACK_FREQUENCY_DURATION_MS 0
#define LCD_FEEDBACK_FREQUENCY_HZ 0

#define FILAMENT_CHANGE_ALERT_BEEPS          0

**Comment by Jyers** [comment, marked as answer]
That should be most of them, I'll probably modify the audio feedback for saving and such to remove the beeps as well when they are disabled.

PS - Please pat your dog on my behalf as an apology for the excessive beeps

**Reply by embraceentropy** [reply]
Thank you! You're already my dogs friend for eliminating 95% of those beeps that, apparently, are here to kill us.

**Reply by AngleV** [reply]
There is also another fix for your dogs sensitive ears.
Unscrew the lcd find the buzzer and put a tape on it. Problem solved 🙂

**Reply by Tuncay-Ayhan** [reply]
> There is also another fix for your dogs sensitive ears.
> Unscrew the lcd find the buzzer and put a tape on it. Problem solved 🙂

@embraceentropy 

### Well, if you are willing to got that far. I'll raise you with

1. Check the orientation off the piezo before hand. Desolder the piezo speaker. 
2. Get an LED, 2 wires (diff. color ). Approx. 2x 5cm. A 100ohm / 200ohm resistor. 
![diode](https://user-images.githubusercontent.com/28215053/117874973-eec07080-b2a1-11eb-9ef8-244ce347a010.jpg)
3. Measure the diameter of the LED
4. Get a drill
5. Drill a little hole on the bottom of the LCD case.
![photo_2021-05-11_21-38-02](https://user-images.githubusercontent.com/28215053/117874810-c2a4ef80-b2a1-11eb-9931-95b4046af751.jpg)
6. Don't forget to check the polarity again. Wire everything up. 
7. Done. 

This should do the job.


p.s. 

I did the tape trick a few weeks ago 😂

**Reply by AngleV** [reply]
https://youtu.be/hc2TLJFKazs
🤣🤣🤣

**Reply by Tuncay-Ayhan** [reply]
> https://youtu.be/hc2TLJFKazs
> 🤣🤣🤣

Haha, I love that song. It's so cringe.

**Comment by embraceentropy** [comment]
@Jyers Thank you for the update removing more of the LCD beeps! The only beeps I get now are during the filament change when I'm being prompted by LCD to insert filament. This occurs when using the change filament feature on the LCD and if I send M600 directly to the printer. Is there something I can change in the code to disable the beeps during filament change?

**Reply by Jyers** [reply]
Looks like a better implementation is going to have to wait until I can better integrate the code into MarlinUI. If you want a fix for now, go into Marlin\src\lcd\marlinui.h and change `static constexpr bool buzzer_enabled = true;` to `static constexpr bool buzzer_enabled = false;`

**Reply by embraceentropy** [reply]
@Jyers Thank you! This worked! I definitely understand why it isn't in the repo, so no worries. I'm probably one of the few who wouldn't want a beep reminder during M600. Thanks again!

**Comment by Cal-LF** [comment]
@embraceentropy May I request of you to offer your Bunny Icon Set to Jyres display firmware? I like it and I think others will like your set as another option.

**Reply by embraceentropy** [reply]
@Cal-LF Sure! I can't take credit for anything as I only combined the two other packs in the repo. What do I need to do?

**Reply by Cal-LF** [reply]
@embraceentropy I will post it here.
@Jyers This Icon pack is nice. Could we add it to the display firmware?
[Bunny Icon Set.zip](https://github.com/Jyers/Marlin/files/6535492/Bunny.Icon.Set.zip)

**Reply by Jyers** [reply]
Yeah, sure.
I'll add it soon

**Reply by gothcha** [reply]
This Bunny set is actually the set I use, but when I submitted it I only included the changed icons so the earlier modified icons were not included in the published set. So I agree this is the best complete set of custom icons. Extra note it includes ICON 206 the Probe icon which requires code changes to enable.

Maybe just use this in place of the existing Gotcha pack

Just a thought, @embraceentropy your 9.ICO may be too large 228K? I had to reduce the size to get the last few 251-254 manual move icons to load correctly.

@Jyers for the sake of completeness here is my full icon set including all the files and the DWIN_SET with the 9.ICO within the file size limits. Could you replace the existing Gotcha sets with these, and thanks to @embraceentropy for spotting that it was not complete in the repo.

[Gotcha's Pack.zip](https://github.com/Jyers/Marlin/files/6538118/Gotcha.s.Pack.zip)
[DWIN_SET (Gotcha).zip](https://github.com/Jyers/Marlin/files/6538121/DWIN_SET.Gotcha.zip)

**Reply by embraceentropy** [reply]
@gothcha I was able to flash on my 9.ico to the LCD without issue (that I know of, at least haha).

**Reply by embraceentropy** [reply]
@gothcha I was able to flash on my 9.ico to the LCD without issue (that I know of, at least haha).

**Reply by Cal-LF** [reply]
@embraceentropy I looked closely at both icon sets on the printer screen. They appear to be complete and none missing.

**Reply by gothcha** [reply]
Go to manual levelling and check the icons for top left, bottom right etc. These may be missing if the file size is exceeded. I and @Jyers had this issue when we extended the icon set.

**Reply by gothcha** [reply]
@Cal-LF They are not missing as such, the pack just had an odd mix of stock and custom!

**Reply by Jyers** [reply]
> Could you replace the existing Gotcha sets with these

@gothcha will do! Thanks for clearing that up
